service/grpc/topic.go

```diff
@@ -117,6 +118,20 @@ func (s *Server) OnTopicEvent(ctx context.Context, in *runtimev1pb.TopicEventReq
 			}
 		}
 
+		// extract custom metadata from the context
+		var md map[string]string
+		meta, ok := metadata.FromIncomingContext(ctx)
+		if ok {
+			for k, v := range meta {
+				if strings.HasPrefix(strings.ToLower(k), "metadata.") {
+					if md == nil {
+						md = make(map[string]string)
```

Comment:
```suggestion
						md = make(map[string]string)
```
Instead of checking and assigning here can we assign it at the top itself